django_socio_grpc/decorators.py

```diff
@@ -79,3 +104,217 @@ def wrapper(function):
+                if request_setter:
+                    for key, value in request_setter.items():
+                        setattr(context, key, value)
+                # INFO - AM - 01/08/2024 - Give the HTTP decorator a wrapper around the endpoint that match what a django endpoint should expect as param and return type
+                response_proxy = decorator_to_wrap(_simulate_function)(
+                    service_instance, request, context
+                )
+                return response_proxy.grpc_response
+
+        return _view_wrapper
+
+    return decorator
+
+
+def vary_on_metadata(*headers) -> Callable:
+    """
+    Same as https://github.com/django/django/blob/0e94f292cda632153f2b3d9a9037eb0141ae9c2e/django/views/decorators/vary.py#L8
+    but need to wrap the response in a GRPCInternalProxyResponse.
+    A view decorator that adds the specified metadatas to the Vary metadata of the
+    response. Usage:
+
+        @vary_on_metadata('Cookie', 'Accept-language')
+        def index(request):
+            ...
+
+    Note that the metadata names are not case-sensitive.
+    """
+    return http_to_grpc(vary_on_headers(*headers), support_async=True)
+
+
+@functools.wraps(cache_page)
+def cache_endpoint(*args, **kwargs):
+    """
+    This decorator is an helper to used the cache_page decorator from django on a grpc endpoint.
+    Please to not use on Create, Update or Delete endpoint as it will cache the response and return the same response to all the user.
+
+    It's step are:
+    - Transform cache_page to a method decorator see https://docs.djangoproject.com/en/5.0/topics/class-based-views/intro/#decorating-the-class
+    - Transform the cache_page method decorator to a grpc compatible decorator
+    - Specify that we need to force the request to be a GET request
+    """
+    return http_to_grpc(
+        method_decorator(cache_page(*args, **kwargs)),
+        request_setter={"method": "GET"},
+        support_async=False,
+    )
+
+
+def cache_endpoint_with_deleter(
+    timeout: int,
+    key_prefix: str,
+    senders: Iterable[Model],
+    cache: str = None,
+    invalidator_signals: Iterable[Callable] = None,
+):
+    """
+    This decorator do all the same as cache_endpoint but with the addition of a cache deleter.
+    The cache deleter will delete the cache when a signal is triggered.
+    This is useful when you want to delete the cache when a model is updated or deleted.
+
+    :param timeout: The timeout of the cache
+    :param key_prefix: The key prefix of the cache
+    :param cache: The cache alias to use. If None, it will use the default cache. It is named cache and not cache_alias to keep compatibility with Django cache_page decorator
+    :param senders: The senders to listen to the signal
+    :param invalidator_signals: The django signals to listen to delete the cache
+    """
+    if not key_prefix:
+        logger.warning(
```

Comment:
Because if not using Redis there is no way to clean multiple key from cache without knowing them exactly.
So we clear all the cach: https://github.com/socotecio/django-socio-grpc/pull/313/files#diff-e320866935eaa353b47eb8ff96e0bbcb20aa7fa3b0ae29b08681b63604816cb6R308